packages/story-editor/src/components/library/panes/media/local/hotlink/hotlinkModal.js

```diff
@@ -0,0 +1,194 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * External dependencies
+ */
+import { __, sprintf, translateToExclusiveList } from '@web-stories-wp/i18n';
+import { Input } from '@web-stories-wp/design-system';
+import { useState, useCallback } from 'react';
+import styled from 'styled-components';
+import {
+  createResource,
+  getFileNameFromUrl,
+  getFirstFrameOfVideo,
+  getImageDimensions,
+  getVideoDimensions,
+  hasVideoGotAudio,
+  getVideoLength,
+} from '@web-stories-wp/media';
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import { useConfig } from '../../../../../../app';
+import Dialog from '../../../../../dialog';
+import useLibrary from '../../../../useLibrary';
+import {
+  useUploadVideoFrame,
+  getPosterName,
+} from '../../../../../../app/media/utils';
+import { withProtocol } from '../../../../../../utils/url';
+
+const InputWrapper = styled.div`
+  margin: 16px 4px;
+  width: 470px;
+  height: 100px;
+`;
+
+function HotlinkModal({ isOpen, onClose }) {
```

Comment:
Should we have a storybook story for this component like we do for other dialogs?

Reply:
Sure! I thought that we might remove the modal with the final UX but it's a small thing to do, so no harm adding now.